File: WikiCodeParser/Nodes/NodeCollection.cs
using System.Collections.Generic;
using System.Linq;

namespace WikiCodeParser.Nodes
{
    /// <summary>
    /// A list of nodes.
    /// </summary>
    public class NodeCollection : INode
    {
        public List<INode> Nodes { get; set; }

        public NodeCollection(params INode[] nodes) : this(nodes.AsEnumerable())
        {
            //
        }

        public NodeCollection(IEnumerable<INode> nodes)
        {
            Nodes = nodes.ToList();
        }

        public string ToHtml() => string.Join("", Nodes.Select(x => x.ToHtml()));
        public string ToPlainText() => string.Join("", Nodes.Select(x => x.ToPlainText()));
        public IList<INode> GetChildren() => Nodes;
        public void ReplaceChild(int i, INode node) => Nodes[i] = node;
    }
}

File: src/Nodes/NodeCollection.ts
import { INode } from './INode';

export class NodeCollection implements INode {
    public Nodes : INode[];

    constructor(...nodes : INode[]) {
        this.Nodes = Array.from(nodes);
    }

    ToHtml(): string {
        return this.Nodes.map(x => x.ToHtml()).join('');
    }
    ToPlainText(): string {
        return this.Nodes.map(x => x.ToPlainText()).join('');
    }
    GetChildren(): INode[] {
        return this.Nodes;
    }
    ReplaceChild(i: number, node: INode): void {
        this.Nodes[i] = node;
    }
    HasContent(): boolean {
        return this.Nodes.some(x => x.HasContent());
    }
}